src/main/java/com/amazonaws/services/kinesis/clientlibrary/lib/worker/Worker.java

```diff
@@ -673,30 +687,29 @@ private void initialize() {
                 LOG.info("Initializing LeaseCoordinator");
                 leaseCoordinator.initialize();
 
-                TaskResult result = null;
-                if (!skipShardSyncAtWorkerInitializationIfLeasesExist
-                        || leaseCoordinator.getLeaseManager().isLeaseTableEmpty()) {
-                    LOG.info("Syncing Kinesis shard info");
-                    ShardSyncTask shardSyncTask = new ShardSyncTask(streamConfig.getStreamProxy(),
-                            leaseCoordinator.getLeaseManager(), initialPosition, cleanupLeasesUponShardCompletion,
-                            config.shouldIgnoreUnexpectedChildShards(), 0L, shardSyncer, null);
-                    result = new MetricsCollectingTaskDecorator(shardSyncTask, metricsFactory).call();
-                } else {
-                    LOG.info("Skipping shard sync per config setting (and lease table is not empty)");
+                if (shouldInitiateLeaseSync()) {
+                    LOG.info(config.getWorkerIdentifier() + " worker is beginning initial lease sync.");
+                    TaskResult result = leaderElectedPeriodicShardSyncManager.syncShardsOnce();
+                    if (result.getException() != null) {
```

Comment:
I believe this code path will never be executed since we never propagate the exception initially

Reply:
The exception gets propagated within the TaskResult in metricsEmittingShardSyncTask.call() (TaskResult) -> runShardSync (TaskResult) -> syncShardsOnce (TaskResult). So any exception in the shard sync task will make it here. I'll add a unit test to validate.

Reply:
Actually, this behavior is already validated in `testInitializationFailuresWithRetries()`. KinesisProxy is mocked to throw on calling the service, and KCL retries initialization attempts, meaning this code was exercised.